src/openklant/components/klantinteracties/api/validators.py

```diff
@@ -166,3 +171,27 @@ def Rekeningnummer_exists(value):
         Rekeningnummer.objects.get(uuid=str(value))
     except Rekeningnummer.DoesNotExist:
         raise serializers.ValidationError(_("Rekeningnummer object bestaat niet."))
+
+
+class OptionalEmailValidator(EmailValidator):
+    """
+    EmailValidator for SoortDigitaalAdres that only attempts to validate if
+    `SoortDigitaalAdres` is `email`
+    """
+
+    requires_context = True
+
+    def __call__(
+        self, value: str, serializer_field: serializers.Field, attrs: dict | None = None
+    ):
+        if serializer_field:
+            serializer = serializer_field.parent
+            serializer = cast(serializers.Serializer, serializer)
+            soort_digitaal_adres = get_field_value(
+                serializer, serializer.initial_data, "soort_digitaal_adres"
+            )
+        else:
+            soort_digitaal_adres = attrs.get("soort_digitaal_adres")
```

Comment:
A little nitpicking: this part doesn't actually provide logic for OptinalEmailValidator, it's more about how to retrieve `soort_digitaal_adres`. I'd move this part to serializer.validate_adres and form clean_adres methods.
And the validator itself would have parameters: 
```
    def __call__(self, value: str,  soort_digitaal_adres:str):
```

But it really is a matter of preference so please treat my comment as optional

Reply:
You're right, initially I added the validator to `validators` on the serializer.Meta, that's why I had this setup here. But since I manually call the validator now in `.validate`, it makes more sense to do it like you suggested